tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/jvmmodel/TclJvmTypeReferenceUtil.xtend

```diff
@@ -0,0 +1,192 @@
+		}
+	}
+	
+	private def ensureTypeReferenceBuilderInitialized() {
+		if (typeReferenceBuilder === null) {
+			throw new RuntimeException("Please call initWith beforehand!")
+		}
+	}
+	
+	def JvmTypeReference buildFrom(Class<?> clazz, JvmTypeReference ... typeArgs) { // allow varargs (which is not supported by create methods, or so it seems)
+		return _buildFrom(clazz, typeArgs)
+	}
+
+	private def JvmTypeReference create typeReferenceBuilder.typeRef(clazz, typeArgs) _buildFrom(Class<?> clazz, JvmTypeReference[] typeArgs) {
+		if (it instanceof JvmUnknownTypeReference) {
+			throw new RuntimeException('''Cannot create reference for type '«clazz.name»', initWith was probably called with wrong resourceset.''')
+		} 
+	}
+
+	def boolean isJsonArray(JvmTypeReference typeReference) {
+		jsonArrayJvmTypeReference.qualifiedName == typeReference?.qualifiedName // let's see if this works
+	}
+
+	def boolean isJsonObject(JvmTypeReference typeReference) {
+		jsonObjectJvmTypeReference.qualifiedName == typeReference?.qualifiedName
+	}
+
+	def boolean isJsonPrimitive(JvmTypeReference typeReference) {
+		jsonPrimitiveJvmTypeReference.qualifiedName == typeReference?.qualifiedName
+	}
+
+	def boolean isJsonElement(JvmTypeReference typeReference) {
+		jsonElementJvmTypeReference.qualifiedName == typeReference?.qualifiedName
+	}
+
+	def boolean isString(JvmTypeReference typeReference) {
+		return String.name == typeReference?.qualifiedName
+	}
+
+	def boolean isBoolean(JvmTypeReference typeReference) {
+		val qname = typeReference?.qualifiedName
+		return (Boolean.name == qname || boolean.name == qname)
+	}
+
+	def boolean isLong(JvmTypeReference typeReference) {
+		val qname = typeReference.qualifiedName
+		return (Long.name == qname || long.name == qname)
+	}
+
+	def boolean isJson(JvmTypeReference typeReference) {
+		return jsonUtil.isJsonType(typeReference)
+	}
+
+	def boolean isOrderable(JvmTypeReference typeReference) {
+		return isLong(typeReference) // currently the only one known to be orderable is long
+	}
+	
+	// use xbase to check for assignablility
+	def boolean isAssignableFrom(JvmTypeReference target, JvmTypeReference source,
+		TypeConformanceComputationArgument argument) {
+		if (typeReferenceOwner === null) {
+			typeReferenceOwner = new StandardTypeReferenceOwner(services, null as ResourceSet)
+		}
+		if (target.qualifiedName == source.qualifiedName) { // workaround for a bug, where two types that boil down to java.lang.String were deemed NOT assignable
+			return true
+		}
+		val lleft = typeReferenceOwner.toLightweightTypeReference(target)
+		var lright = typeReferenceOwner.toLightweightTypeReference(source)
+		val assignable = lleft.isAssignableFrom(lright, argument)
+		return assignable
+	}
+
+	def boolean isAssignableFrom(JvmTypeReference target, JvmTypeReference source) {
+		return isAssignableFrom(target, source, new TypeConformanceComputationArgument)
+	}
+	
+	def isNonFractionalNumber(JvmTypeReference reference) {
```

Comment:
`isLongOrInt` would be more clear. Also the return type is missing (same with isInt and isLong).